refactor: Add unit tests for new GroupChannelListProvider

git-babel opened the pull request:
This PR contains the unit tests for the recent refactoring of `GroupChannelListProvider`.
* add test about the new provider and its state management logic itself.
* add the integration test with `GroupChannelListUI` component.

git-babel commented:
I have a question. We have some lines like below in `GroupChannelListProvider`
```javascript
  const channelListDataSource = useGroupChannelList(sdk, {
    collectionCreator: getCollectionCreator(sdk, channelListQueryParams),
    markAsDelivered: (channels) => channels.forEach(scheduler.push),
    onChannelsDeleted: (channelUrls) => {
      channelUrls.forEach((url) => {
        if (url === selectedChannelUrl) onChannelSelect(null);
      });
    },
  });
  const { refreshing, initialized, groupChannels, refresh, loadMore } = channelListDataSource;
```
that `useGroupChannelList` comes from `@sendbird/uikit-tools` package. I'm not sure that we have to include those state from `channelListDataSource` in our unit test.
It seems like that the `sendbird/uikit-tools` already handle the state updates. And it was hard to call some appropriate action and observe the state change cause those logic are inside of `@sendbird/uikit-tools`. For now, those states from `channelListDataSource` is excluded in this PR's test cases. Should I add them?

HoonBaek commented on src/modules/GroupChannelList/context/GroupChannelListProvider.tsx:
```diff
@@ -66,12 +65,12 @@ const initialState: GroupChannelListState = {
-  onCreateChannelClick: noop,
-  onBeforeCreateChannel: null,
-  onUserProfileUpdated: noop,
+  onChannelSelect: undefined,
+  onChannelCreated: undefined,
+  onThemeChange: undefined,
+  onCreateChannelClick: undefined,
+  onBeforeCreateChannel: undefined,
+  onUserProfileUpdated: undefined,
```
We don't handle the undefined functions on the other files, like optional chaining.

We should confirm if there's no Function undefined error, during run-time.
or use `noop` again

git-babel replied:
This is updated like the comment below by Irene.

HoonBaek commented on src/modules/GroupChannelList/context/GroupChannelListProvider.tsx:
```diff
@@ -191,7 +193,7 @@ export const GroupChannelListManager: React.FC<GroupChannelListProviderProps> =
     refreshing,
     initialized,
-    groupChannels,
+    // groupChannels,
```
Is this a typo?

git-babel replied:
Fixed.

AhyoungRyu commented:
Just checked the test coverage and it seems the newly created `useGroupChannelList` file doesn't have corresponding test file. Could you add one too? Also, I hope the coverage of  `GroupChannelListProvider` can be green not yellow. 
- Before this change 
<img width="890" alt="Screenshot 2024-10-21 at 2 57 41 PM" src="https://github.com/user-attachments/assets/5be967cb-1a86-446e-b97e-862222b32d3b">

- After this change 
<img width="988" alt="Screenshot 2024-10-21 at 2 59 56 PM" src="https://github.com/user-attachments/assets/4bc0782f-aec2-4a1e-94b8-9359a60e8af6">


You can also get some hint from here https://github.com/sendbird/sendbird-uikit-react/pull/1234#issuecomment-2425652104 about the way you increase the coverage. It can be checked with `yarn test:coverage` command on the root of this project.

AhyoungRyu commented:
>For now, those states from channelListDataSource is excluded in this PR's test cases. Should I add them?

Don't need to add for it since the logic is from mostly outside of the main logic we want to test.

git-babel commented on src/modules/GroupChannelList/context/GroupChannelListProvider.tsx:
```diff
@@ -191,7 +193,7 @@ export const GroupChannelListManager: React.FC<GroupChannelListProviderProps> =
     refreshing,
     initialized,
-    groupChannels,
+    // groupChannels,
```
Applied.

git-babel commented on src/modules/GroupChannelList/context/__tests__/GroupChannelListProvider.spec.js:
I changed it to '.tsx'

git-babel commented on src/modules/GroupChannelList/context/__tests__/useGroupChannelList.spec.js:
I changed it to '.tsx'

git-babel commented:
I added and updated the tests for new provider and hook. The result is like below.
<img width="1063" alt="스크린샷 2024-10-23 오전 11 07 19" src="https://github.com/user-attachments/assets/280c0955-b9a4-48e0-aea4-4314807960a4">
The unvisited lines in `GroupChannelListProvider` are the callback functions to be called in `@sendbird/uikit-tools`. The actual implementation of functions from `@sendbird/uikit-tools` is shadowed by the mocks in test file, because originally it works only when the valid data is given. So it is hard to cover those lines now.

AhyoungRyu commented on src/modules/GroupChannelList/context/GroupChannelListProvider.tsx:
```diff
@@ -87,27 +87,26 @@ const useGroupChannelListStore = () => {
-    isTypingIndicatorEnabled,
-    isMessageReceiptStatusEnabled,
+  disableAutoSelect = false,
+  allowProfileEdit = false,
+  isTypingIndicatorEnabled = false,
+  isMessageReceiptStatusEnabled = false,
```
```suggestion
  allowProfileEdit,
  isTypingIndicatorEnabled,
  isMessageReceiptStatusEnabled,
```
We'd better to keep the original status of this prop as-is since there might be a chance that some behavior changes due to this default value setting.

git-babel replied:
Applied.

HoonBaek reviewed (approved):
👍
